bats/index.bats

```diff
@@ -2316,3 +2316,89 @@ SQL
+  FOREIGN KEY (parent_value)
+    REFERENCES parent(v1)
+);
+CREATE TABLE child_unq (
+  pk varchar(10) PRIMARY KEY,
+  parent_value bigint,
+  FOREIGN KEY (parent_value)
+    REFERENCES parent(v1)
+);
+CREATE TABLE child_non_unq (
+  pk varchar(10) PRIMARY KEY,
+  parent_value bigint,
+  FOREIGN KEY (parent_value)
+    REFERENCES parent(v1)
+);
+INSERT INTO parent VALUES (1, 1), (2, 2), (3, 3), (4, NULL), (5, 5), (6, 6), (7, 7);
+INSERT INTO child VALUES ('1', 1), ('2', NULL), ('3', 3), ('4', 3), ('5', 5);
+INSERT INTO child_idx VALUES ('1', 1), ('2', NULL), ('3', 3), ('4', 3), ('5', 5);
+INSERT INTO child_unq VALUES ('1', 1), ('2', NULL), ('3', 3), ('4', NULL), ('5', 5);
+INSERT INTO child_non_unq VALUES ('1', 1), ('2', NULL), ('3', 3), ('4', 3), ('5', 5);
+SQL
+    dolt sql -q "CREATE INDEX abc ON child (parent_value);"
+    run dolt sql -q "CREATE INDEX abc_idx ON child_idx (parent_value);"
+    [ "$status" -eq "1" ]
+    [[ "$output" =~ "duplicate" ]] || false
+    dolt sql -q "CREATE UNIQUE INDEX abc_unq ON child_unq (parent_value);"
+    run dolt sql -q "CREATE UNIQUE INDEX abc_non_unq ON child_non_unq (parent_value);"
+    [ "$status" -eq "1" ]
+    [[ "$output" =~ "UNIQUE constraint violation" ]] || false
+
+    run dolt schema show child
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ 'KEY `abc` (`parent_value`)' ]] || false
+    ! [[ "$output" =~ 'KEY `parent_value` (`parent_value`)' ]] || false
+    run dolt schema show child_idx
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ 'KEY `parent_value` (`parent_value`)' ]] || false
+    ! [[ "$output" =~ 'KEY `abc_idx` (`parent_value`)' ]] || false
+    run dolt schema show child_unq
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ 'KEY `abc_unq` (`parent_value`)' ]] || false
+    ! [[ "$output" =~ 'KEY `parent_value` (`parent_value`)' ]] || false
+    run dolt schema show child_non_unq
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ 'KEY `parent_value` (`parent_value`)' ]] || false
+    ! [[ "$output" =~ 'KEY `abc_non_unq` (`parent_value`)' ]] || false
+
+    run dolt sql -q "SELECT * FROM child WHERE parent_value = 5" -r=csv
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ "pk,parent_value" ]] || false
+    [[ "$output" =~ "5,5" ]] || false
+    [[ "${#lines[@]}" = "2" ]] || false
+    run dolt sql -q "SELECT * FROM child_idx WHERE parent_value = 5" -r=csv
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ "pk,parent_value" ]] || false
+    [[ "$output" =~ "5,5" ]] || false
+    [[ "${#lines[@]}" = "2" ]] || false
+    run dolt sql -q "SELECT * FROM child_unq WHERE parent_value = 5" -r=csv
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ "pk,parent_value" ]] || false
+    [[ "$output" =~ "5,5" ]] || false
+    [[ "${#lines[@]}" = "2" ]] || false
+    run dolt sql -q "SELECT * FROM child_non_unq WHERE parent_value = 5" -r=csv
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ "pk,parent_value" ]] || false
+    [[ "$output" =~ "5,5" ]] || false
+    [[ "${#lines[@]}" = "2" ]] || false
+}

```

Comment:
You also want tests that the opposite order works: when you create an index first, then add a foreign key constraint. Everything should work the same.

Reply:
This is in this test.
```sql
CREATE TABLE child_idx (
  pk varchar(10) PRIMARY KEY,
  parent_value bigint,
  INDEX (parent_value),
  FOREIGN KEY (parent_value)
    REFERENCES parent(v1)
);
```
When they're both declared in a `CREATE TABLE`, we declare the indexes first so that the foreign keys that are also declared will reference them.